core/internal/api/api_test.go

```diff
@@ -122,60 +121,13 @@ func newClient(
 	credentialProvider, err := api.NewCredentialProvider(settings)
 	require.NoError(t, err)
 
-	backend := api.New(api.BackendOptions{BaseURL: baseURL,
-		CredentialProvider: credentialProvider})
+	backend := api.New(api.BackendOptions{
+		BaseURL:            baseURL,
+		CredentialProvider: credentialProvider,
+	})
 	return backend.NewClient(opts)
 }
 
-type RequestCopy struct {
-	Method string
-	URL    *url.URL
-	Body   string
-	Header http.Header
-}
-
-type RecordingServer struct {
-	sync.Mutex
-	*httptest.Server
-
-	requests []RequestCopy
-}
-
-// All requests recorded by the server.
-func (s *RecordingServer) Requests() []RequestCopy {
-	s.Lock()
-	defer s.Unlock()
-	return slices.Clone(s.requests)
-}
-
-// Returns a server that records all requests made to it.
-func NewRecordingServer() *RecordingServer {
-	rs := &RecordingServer{
-		requests: make([]RequestCopy, 0),
-	}
-
-	rs.Server = httptest.NewServer(http.HandlerFunc(
-		func(w http.ResponseWriter, r *http.Request) {
-			body, _ := io.ReadAll(r.Body)
-
-			rs.Lock()
-			defer rs.Unlock()
-
-			rs.requests = append(rs.requests,
-				RequestCopy{
-					Method: r.Method,
-					URL:    r.URL,
-					Body:   string(body),
-					Header: r.Header,
-				})
-
-			_, _ = w.Write([]byte("OK"))
-		}),
-	)
-
-	return rs
-}
```

Comment:
I moved this to the apitest file, it seems like a useful thing to use across tests